### PR TITLE
fix-issue-2963

### PR DESCRIPTION
**Pull request by michalguspiel**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

Describe with a few bullets **what's new:**
-  migrated `WalletAccountLogic` to use new `Transaction` class
- Created new `FromToTimeRange` functions that operate on new `Transaction` class
- Created new `WalletCategoryLogic` functions that operate on new `Transaction` class
- new `PlannedPaymentsLogic.payOrGet` functions that operates on new `Transaction` class
- Created new `DueSection` which replaces old `LegacyDueSection `- Usage of it is blocked for now though.
- Removed commented code
- Added more Extension functions to `Transaction` class
- Fixed TransactionMapper

## Risk Factors

**What may go wrong if we merge your PR?**

Not much, keep in mind a lot of these changes are a prerequisite to follow-up refactoring. A strategy to migrate TransactionHistoryDateDivider and TransactionHistoryItem is needed to proceed with follow-up refactoring that concerns the presentation layer. Otherwise incompatibility of new `Transaction` and `TransactionHistoryItem` prevents this refactoring from happening.

## Does this PR closes any GitHub Issues?
Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.
- Relates to #2963

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: shared/data/src/main/java/com/ivy/data/repository/mapper/TransactionMapper.kt [hunk 1]
```diff
@@ -34,6 +35,7 @@ class TransactionMapper @Inject constructor() {
         )
         val settled = dateTime != null
         val time = dateTime?.atZone(zoneId)?.toInstant()
+            ?: dueDate?.atZone(zoneId)?.toInstant()
```
Comment (michalguspiel) [inline shared/data/src/main/java/com/ivy/data/repository/mapper/TransactionMapper.kt:38]: This was preventing planned TransactionEntity to be mapped to Transaction

File: shared/data/src/main/java/com/ivy/data/model/Transaction.kt [hunk 2]
```diff
@@ -89,10 +91,36 @@ fun Transaction.getAccountId(): UUID =
...
 fun Transaction.getTransactionType(): TransactionType {
     return when (this) {
         is Expense -> TransactionType.EXPENSE
         is Income -> TransactionType.INCOME
         is Transfer -> TransactionType.TRANSFER
     }
+}
+
+fun Transaction.settledTransaction(): Transaction {
+    val timeNow = LocalDateTime.now(ZoneOffset.UTC).toInstant(ZoneOffset.UTC)
```
Comment (ILIYANGERMANOV) [inline shared/data/src/main/java/com/ivy/data/model/Transaction.kt:109]: Use `Instant.now()`

File: shared/data/src/main/java/com/ivy/data/model/Transaction.kt [hunk 3]
```diff
@@ -89,10 +91,36 @@ fun Transaction.getAccountId(): UUID =
...
+
 fun Transaction.getTransactionType(): TransactionType {
     return when (this) {
         is Expense -> TransactionType.EXPENSE
         is Income -> TransactionType.INCOME
         is Transfer -> TransactionType.TRANSFER
     }
+}
+
+fun Transaction.settledTransaction(): Transaction {
```
Comment (ILIYANGERMANOV) [inline shared/data/src/main/java/com/ivy/data/model/Transaction.kt:108]: rename to `.settleNow()`

File: shared/data/src/main/java/com/ivy/data/model/Transaction.kt [hunk 4]
```diff
@@ -89,10 +91,36 @@ fun Transaction.getAccountId(): UUID =
         is Transfer -> fromAccount.value
     }
 
+fun Transaction.getAccount(): AccountId = when (this) {
```
Comment (ILIYANGERMANOV) [inline shared/data/src/main/java/com/ivy/data/model/Transaction.kt:94]: Ideally, we'd like to get rid of all of these functions and have only the data" in the model but it's fine for now. Can you move these is some other/package file called migration utils or sth. The reason for that is that I don't want to couple data with behavior. 

Data should remain pure data and open to any interpretation (behavior). So just move these in some other file + package. Maybe create an `temp.migration` package and inside it `TempMigrationUtils.kt`

File: temp/legacy-code/src/main/java/com/ivy/legacy/data/DueSection.kt [hunk 5]
```diff
@@ -5,9 +5,17 @@ import com.ivy.base.legacy.Transaction
...
 @Immutable
-data class DueSection(
+data class LegacyDueSection(
     val trns: ImmutableList<Transaction>,
     val expanded: Boolean,
     val stats: IncomeExpensePair
 )
+
+@Immutable
+data class DueSection(
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/data/DueSection.kt:17]: Please, keep the deprecation - ideally, we'd like to get rid of these UI layer models and replace them with more optimal ones

**Review by ILIYANGERMANOV — COMMENTED** [review]
Nice work 💯 Overall LGTM! Left some comments. Also, please make sure to keep deprecates and add `@Deprecated()` annotations to all legacy data structures.

After all migrations are completed we'll have:
- `shared:data`: Data layer with the data model
- `:shared:domain`: Use-cases using the Data Layer and replacing all legacy `*Logic`, `Actions` and stuff from the `:temp:*` modules
- `:shared:ui`: Shared UI layer with new viewState models optimized for Compose that provide common components like the transactions list, account card ,etc.

File: shared/data/src/main/java/com/ivy/data/model/Transaction.kt (hunk 2 above)
Comment (michalguspiel) [reply shared/data/src/main/java/com/ivy/data/model/Transaction.kt]: Done

File: shared/data/src/main/java/com/ivy/data/model/Transaction.kt (hunk 3 above)
Comment (michalguspiel) [reply shared/data/src/main/java/com/ivy/data/model/Transaction.kt]: Done

File: temp/legacy-code/src/main/java/com/ivy/legacy/data/DueSection.kt (hunk 5 above)
Comment (michalguspiel) [reply temp/legacy-code/src/main/java/com/ivy/legacy/data/DueSection.kt:18]: Done

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/data/DueSection.kt:18]: @michalguspiel just fyi when you're ready for re-review click the ♻️ refresh button next to my regiew. Also feel free to resolve all fixed comments

**Pull request merged**